lib/index.js

```diff
@@ -47,8 +49,190 @@ api.validateConfiguration = (validatorConfig, callback) =>
+      'Unsupported signature type.',
+      'NotSupportedError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }
+
+  const signature = {};
+  proofType.forEach(type => signature[type] = event.signature
+    .filter(s => s.type === type));
+  const hasOneOfEach = proofType.every(type => signature[type].length !== 0);
+
+  // FIXME: the `minimumProofsRequired` property defined now seems inadequate to
+  // indicate that we actually want x number of one sig type, and y number of
+  // another sig type.
+  // For this first use case, we will not accept *only* 2 ld-signatures
+  // or *only* 2 equihash sigs.  `minimumProofsRequired` is currently being
+  // ignored
+
+  if(!hasOneOfEach) {
+    return callback(new BedrockError(
+      'The signature requirements have not been met.',
+      'ValidationError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }
+
+  async.each(signature, (s, callback) => {
+    const sigZero = s[0];
+    const e = bedrock.util.clone(event);
+    e.signature = sigZero;
+    if(sigZero.type === 'LinkedDataSignature2015') {
+      return _verifyLinkedData(e, (err, result) => {
+        if(err) {
+          return callback(err);
+        }
+        if(!result.verified) {
+          callback(new BedrockError(
+            'LinkedData Signature(s) could not be verified.',
+            'ValidationError', {
+              httpStatusCode: 400,
+              public: true,
+              event,
+              keyResults: result.keyResults
+            }));
+        }
+        // success, do not return an error
+        callback();
+      });
+    }
+    if(sigZero.type === 'EquihashProof2017') {
+      return _verifyEquihash(e, {equihashParameterAlgorithm}, (err, result) => {
+        if(err) {
+          return callback(err);
+        }
+        if(!result) {
+          callback(new BedrockError(
+            'Equihash Signature could not be verified.',
+            'ValidationError', {
+              httpStatusCode: 400,
+              public: true,
+              event
+            }));
+        }
+        // success, do not return an error
+        callback();
+      });
+    }
+    // NOTE: this callback is for safety and *should* never be reached
+    callback(new BedrockError(
+      'Unsupported signature type.',
+      'OperationError', {
+        httpStatusCode: 400,
+        public: true,
+        event,
+      }));
+  }, err => {
+    console.log('EEEEEEEE', err);
+    callback(err);
+  });
+}
+
+function _verifyLinkedData(doc, callback) {
+  let cacheKey;
+  jsigs.verify(doc, {
+    getPublicKey: _getPublicKey,
+    getPublicKeyOwner: _getPublicKeyOwner
+  }, (err, result) => {
+    console.log('11111111', err);
+    console.log('22222222', result);
+    callback(err, result);
+  });
+
+  function _getPublicKey(keyId, options, callback) {
+
+    // FIXME: this seems very specific to the veres-one use case?
```

Comment:
What is very specific about it? Note that "authenticationCredential" is going to be the new way to look for "publicKey" in "identity" documents.

Reply:
sure, but this validator is not limited to being used for this *particular* type of ledger right?

Reply:
Solution, this validator is designed to work with identity documents exclusively and perhaps renamed to indicate such.

Reply:
Done.